actionpack/lib/action_dispatch/http/url.rb

```diff
@@ -1,26 +1,36 @@
 # frozen_string_literal: true
 
 require "active_support/core_ext/module/attribute_accessors"
+require "action_dispatch/http/uri"
 
 module ActionDispatch
   module Http
     module URL
-      IP_HOST_REGEXP  = /\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/
-      HOST_REGEXP     = /(^[^:]+:\/\/)?(\[[^\]]+\]|[^:]+)(?::(\d+$))?/
-      PROTOCOL_REGEXP = /^([^:]+)(:)?(\/\/)?$/
+      class << self
+        def tld_length=(tld_length)
```

Comment:
I think these can `delegate .., to: "ActionDispatch::Http::URI"` (unless we want to deprecate)

Reply:
So, what should we do? I would deprecate those methods. Should I go ahead and deprecate those, otherwise delegate is the way to go, that makes sense..

Reply:
added the delegation ..